### ISSUE TITLE
Binding to the events of the media player instance doesnt behave the same on editor and android device

### ISSUE THREAD

**Issue by SplunkyDev** [RenderHeads/UnityPlugin-AVProVideo#1895]
**Describe the issue**
Bind to the MediaPlayer events to check to state of the media actions. The logs printing the event types are not matching. The editor event types and device event types that are being logged are different even though the actions are the same. (by actions I mean Play,Pause, Seek ahead, Seek back)

```
  m_MediaPlayer.Events.AddListener(OnMediaPlayerEvents); //m_MediaPlayer is the reference to the Media Player component in the scene
  
//this is the sample code from the docs, I have used the event type to handle certain functionality. But current just show for debug log 
  private void OnMediaPlayerEvents(MediaPlayer mp, MediaPlayerEvent.EventType eventType, ErrorCode code)
 {
     Debug.Log("[MediaPlayer] " + mp.name + " generated event: " + eventType.ToString());
     if (eventType == MediaPlayerEvent.EventType.Error)
     {
         Debug.LogError("Error: " + code);
     }
}
```

**Your Setup (please complete the following information):**
 - Unity version: 2022.3.31f1
 - AVPro Video version 2.9.3 ultra (also noticed in version 3 FREE)
 - Operating system version: Android 12 and Windows 10(Editor)
 - Device model: Xiaomi Pad 6 
 - Video specs (resolution, frame-rate, codec, file size): 
 Resolution: 1280*720
 Frame-rate: 18fps
 Codec: H264-MPEG-4 AVC
 Container: MP4
 Size: 15MB

**To Reproduce**
Version 2.9.3
1. Play video-> Pause video ->  m_MediaPlayer.Control.SeekToFrameRelative(relativeFrame); //in this example -3 to +3
2. Play video : Notice on the android device Unpause will not be called it calls StartedSeeking and the FinishedSeeking 

Version 3
1. Play video-> Pause video ->  m_MediaPlayer.Control.SeekToFrameRelative(relativeFrame); //in this example -3 to +3
2. While Seeking on the editor StartedSeeking and FinishedSeeking is called but on the device StartedBuffer and FinishedBuffer is called

**Comment by Ste-RH** [comment]
v2.x is now deprecated and we will no longer be investigating/issuing fixes/updates.

Regarding v3.x, we addressed a seeking/finished event issue in v3.0.2.  What version of 3.x are you testing here?

**Comment by SplunkyDev** [comment]
My apologies for the late response.

![image](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/60402652/001f2cf9-a35c-411b-9341-677c5b3a10ae)


As you can see its version 3.0.2.
I have used the MediaPlayer path as you can see in the code snippet shared in the ticket

**Comment by MorrisRH** [comment]
The finished seeking event wasn't being fired correctly on Android. This will be fixed in the next release (version 3.0.4).

**Comment by Chris-RH** [comment]
AVPro Video version 3.0.4 has been released.  Please let us know if it has not fixed your issue.
